js/apps/admin-ui/src/user/UserForm.tsx

```diff
@@ -223,8 +224,8 @@ export const UserForm = ({
           <FederatedUserLink user={user} />
         </FormGroup>
       )}
-      {isUserProfileEnabled && user?.userProfileMetadata ? (
-        <UserProfileFields config={user.userProfileMetadata} />
+      {isUserProfileEnabled ? (
+        <UserProfileFields config={user?.userProfileMetadata || config!} />
```

Comment:
Good catch, I was not aware we had this existing place where this config was stored. I'd still like to keep the condition if possible, as I am not a fan of circumventing the type system.

Perhaps we can do the following:

```tsx
const userProfileConfig = user?.userProfileMetadata ?? config;

{isUserProfileEnabled && userProfileConfig ? (
  <UserProfileFields config={userProfileConfig} />
) : (
  <>{*/ The other stuff */}</>
)}
```

Would that still work?

Reply:
thanks @jonkoops yes, this will work! We can use this format.